multidim_threshold/threshold.py

```diff
@@ -0,0 +1,52 @@
+import numpy as np
+from copy as deepcopy
+
+from .refine import box_edges
+from .rectangles import to_rec
+
+# Return False only if the polarity is wrong
+compute_var_polarity = lambda stleval, bot, top: stleval(bot) <= stleval(top)
+
+# Go through each edge fo teh rectangle. Do not revisit and update with an
+# edge's polarity data if that edge has already been set to False
+def compute_polarity(stleval, r):
+    # stleval: an evaluator of the STL formula
+    # r: bounding rectangle
+    pol = np.repeat(True, len(r.top))
+    edges = list(box_edges(r))
+    for edge in edges:
+        edge_index = (np.array(edge.top) - np.array(edge.bot)).argmin()
+        if pol[edge_index]:
+            pol[edge_index] = compute_var_polarity(stleval=stleval,
+                                                   bot=edge.bot, top=edge.top)
+
+    return pol
```

Comment:
Would it be possible for `compute_var_polarity` to return a bit vector with only the `edge_index` potentially flipped?
Or perhaps, just use a bitvector?
That way you could implement this whole function as:
```python
import operator as op
from functools import reduce

def compute_polarity(oracle, r: Rec):
    return reduce(op.or_, map(compute_polarity, box_edges(r)))
```